test-suite-groovy/src/test/groovy/io/micronaut/kafka/docs/seek/rebalance/ProductListener.groovy

```diff
@@ -0,0 +1,44 @@
+package io.micronaut.kafka.docs.seek.rebalance
+
+import io.micronaut.configuration.kafka.ConsumerAware
+import io.micronaut.configuration.kafka.annotation.*
+import io.micronaut.context.annotation.Requires
+import io.micronaut.core.annotation.NonNull
+import io.micronaut.kafka.docs.Product
+import org.apache.kafka.clients.consumer.*
+import org.apache.kafka.common.TopicPartition
+
+@KafkaListener(offsetReset = OffsetReset.EARLIEST)
+@Requires(property = "spec.name", value = "ConsumerRebalanceListenerSpec")
+class ProductListener implements ConsumerRebalanceListener, ConsumerAware {
+
+    List<Product> processed = []
+    private Consumer consumer
+
+    ProductListener(ProductListenerConfiguration config) {
```

Comment:
why the empty constructor?

Reply:
The empty constructor is there because I need `ProductListenerConfiguration` to produce the test products _before_ `ProductListener` rebalances. This dependency ensures that things happen in a deterministic order.